src/Agent.Sdk/ContainerInfo.cs

```diff
@@ -14,13 +15,13 @@ public class ContainerInfo
         private List<MountVolume> _mountVolumes;
         private IDictionary<string, string> _userPortMappings;
         private List<PortMapping> _portMappings;
-        private IDictionary<string, string> _environmentVariables;
+        private Dictionary<string, string> _environmentVariables;
+        private Dictionary<string, string> _pathMappings;
+        private PlatformUtil.OS _imageOS;
 
-#if OS_WINDOWS
-        private Dictionary<string, string> _pathMappings = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
-#else
-        private Dictionary<string, string> _pathMappings = new Dictionary<string, string>();
-#endif
+        public delegate void ImageOSChangedHandler();
```

Comment:
It's a little unusual for an event handler not to take any arguments. Typically, you would pass the "sender" as an object in the first parameter and something derived from EventArgs in the second parameter. For this event, it seems like the event args would tell you what OS you are changing to.